second_classes/exercises.livemd

```diff
@@ -51,3 +125,179 @@ graph TB;
 ```
+
+## Setup
+
+```elixir
+defmodule Wordle.Data do
+  def all_possible_guesses do
+    [
+      "starn",
+      "bindi",
+      "woops",
+      "fanos",
+      "cabin",
+      "souct",
+      "trass",
+      "shoat",
+      "lefty",
+      "durra",
+      "hypes",
+      "junta",
+      "baisa",
+      "bises",
+      "kipps",
+      "sable",
+      "abacs",
+      "thurl",
+      "nurrs",
+      "saris",
+      "wroth",
+      "venal",
+      "texas",
+      "soman",
+      "linds",
+      "laden",
+      "nolos",
+      "pixie",
+      "calms",
+      "chert",
+      "oxbow",
+      "groma",
+      "nomen",
+      "potae",
+      "noyed",
+      "fifty",
+      "muils",
+      "pukus",
+      "crane",
+      "tetra",
+      "dwaal",
+      "diels",
+      "ikons"
+    ]
+  end
+
+  def all_possible_words do
+    [
+      "aaaaa",
+      "arbum",
+      "bidoy",
+      "bzfjk",
+      "cqhdw",
+      "dhiyi",
+      "dyksu",
+      "epmng",
+      "fgohs",
+      "fxqce",
+      "gorwq",
+      "hftrc",
+      "hwvlo",
+      "inxga",
+      "jezam",
+      "jwauy",
+      "kncpk",
+      "leejw",
+      "lvgei",
+      "mmhyu",
+      "ndjtg",
+      "nulns",
+      "olnie",
+      "pcpcq",
+      "ptqxc",
+      "qksro",
+      "rbuma",
+      "rswgm",
+      "sjyay",
+      "tazvk",
+      "tsbpw",
+      "ujdki",
+      "vafeu",
+      "vrgzg",
+      "wiits",
+      "wzkoe",
+      "xqmiq",
+      "yhodc",
+      "yypxo",
+      "zprsa"
+    ]
+  end
+end
+```
+
+## Solution
+
+```elixir
+defmodule Wordle.Opening do
+  @moduledoc """
+  Solution:
+  1 - Compute the letter frequency of the list of guessing words
+  2 - Generate all possible words with five letters
+  3 - For each word generated in step two, check its score by
+      seeing how many letters are in the most frequency letters generated in step one
+  4 - Choose the generated word composable by the most frequency letters
+  """
+  alias Wordle.Data
+
+  def start do
+    main_pid = self()
+
+    all_possible_guesses = read_all_possible_guesses()
+    all_possible_words = generate_all_possible_words()
+
+    letters_frequency = compute_frequency(all_possible_guesses)
+
+    Enum.each(all_possible_words, &check_word(&1, letters_frequency, main_pid))
+
+    loop()
+  end
+
+  # TODO: read from wordle_words.txt file
+  defp read_all_possible_guesses, do: Data.all_possible_guesses()
+
+  # TODO: generate all 11_881_376 possible words with five letters
+  # see: https://elixirforum.com/t/how-to-generate-all-possible-3-letter-strings/45574/13
+  defp generate_all_possible_words, do: Data.all_possible_words()
+
+  defp compute_frequency(words) do
+    Enum.reduce(words, %{}, fn word, acc ->
+      word
+      |> String.graphemes()
+      |> Enum.reduce(acc, fn letter, acc ->
+        Map.update(acc, letter, 1, &(&1 + 1))
+      end)
+    end)
+  end
+
+  defp check_word(word, letters_frequency, main_pid) do
+    score =
+      word
+      |> String.graphemes()
+      |> Enum.uniq()
```

Comment:
The fact that the letter is in a word more than once is a piece of extra information. If a guess contains a letter twice we can eliminate all of the words containing a given letter only once, can't we?